internal/server/backgroundjobs.go

```diff
@@ -0,0 +1,77 @@
+package server
+
+import (
+	"context"
+	"time"
+
+	"github.com/infrahq/infra/internal/logging"
+	"github.com/infrahq/infra/internal/server/data"
+	"github.com/infrahq/infra/internal/server/jobs"
+)
+
+// BackgroundJobFunc is the interface for implementing a new background job.
+//
+// currentTime is the time the job was invoked at, and should be used for segmenting records into processable chunks.
+// Jobs may want to ignore records past the currentTime, as those will be handled in a future job run.
+//
+// lastRunAt is the time that the background job was last invoked. This value will only be updated if the function returns a non-error
+//
+// errors will be logged but will not cause the app to crash.
+//
+// panics will be caught and logged
+//
+// jobs should gracefully exit if their context quits, eg ctx.Done() or ctx.Err()
+type BackgroundJobFunc func(ctx context.Context, tx data.WriteTxn, lastRunAt, currentTime time.Time) error
+
+func (s *Server) SetupBackgroundJobs() {
+	s.registerJob(jobs.RemoveOldDeviceFlowRequests, 10*time.Minute)
+	s.registerJob(jobs.RemoveExpiredAccessKeys, 1*time.Minute)
+	s.registerJob(jobs.RemoveExpiredPasswordResetTokens, 15*time.Minute)
+}
+
+func (s *Server) registerJob(job BackgroundJobFunc, every time.Duration) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
The context needs to be passed in, otherwise we can't cancel these goroutines in tests.

```suggestion
func (s *Server) registerJob(ctx context.Context, job BackgroundJobFunc, every time.Duration) {
	ctx, cancel := context.WithCancel(ctx)
```

Reply:
I think we don't need a `stop` function for these, If we change `route.run` to accept a `context.Context`, then we can pass that in directly when it's called.